### PR TITLE
[hotfix][flink-core][doc] add javadoc to improve the code readability

### PR DESCRIPTION
**Pull request by JingGe**
Only added java doc to improve the readability. The code itself did tell enough information wrt the logic. I got understand it via long time debug. It is good to document it for helping developers understand the logic and also for me, since I might forget it after a few months.

No java logic has been changed.

### CODE REVIEW

File: flink-core/src/main/java/org/apache/flink/configuration/Configuration.java [hunk 1]
```diff
@@ -826,6 +836,19 @@ private Optional<Object> getRawValue(String key, boolean canBePrefixMap) {
...
+    /**
+     * This method will do the following steps to get the value of a config option:
+     *
+     * <p>1. get the value from {@link Configuration}. <br>
+     * 2. if key is not found, try to get the value with fallback keys from {@link Configuration}
+     * <br>
+     * 3. if no fallback keys are found, return {@link Optional#empty()}.
+     *
+     * <p>Current logic is not able to get the default value of the fallback key's ConfigOption, if
+     * the given ConfigOption has no default value.
```
Comment (leonardBang) [inline flink-core/src/main/java/org/apache/flink/configuration/Configuration.java:848]: we can mark this as a note although I think this is by design, we get option value by option key firstly and then *try to * fallback to fallback key which is used when  user only configured fallback key. The note is useful when users upgrade from an old flink version which only owned the fallback keys to the higher flink version which deprecated these fallback keys.

**Review by leonardBang — COMMENTED** [review]
Thanks @JingGe for the contribution, I only left one minor comment, maybe we can improve the note better.

**Review by leonardBang — APPROVED** [review]
Thanks @JingGe  for the update , LGTM

**Comment by JingGe** [comment]
@flinkbot run azure

**Pull request merged**